### PR TITLE
refactor(protocol_fee): Remove duplicated code and change `TokenList` data type

### PR DESCRIPTION
**Pull request by notJoon**
## Description

- Remove duplicated helper functions
- Add early return pattern in API functions when token list is empty
- Simplify `DistributeProtocolFee` by removing redundant balance checks and intermediate variables

## Summary by CodeRabbit

# Release Notes

* **Refactor**
  * Improved internal data structure handling for token list management with enhanced consistency checks.
  * Optimized state access patterns to reduce redundant operations.
  * Strengthened error detection for data type inconsistencies.

* **Tests**
  * Added comprehensive test coverage for edge cases and error scenarios.
  * Expanded validation tests for multi-token distributions and state initialization.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: contract/r/gnoswap/protocol_fee/v1/protocol_fee.gno [hunk 1]
```diff
@@ -139,29 +143,29 @@ func (pf *protocolFeeV1) SetDevOpsPct(pct int64) {
...
+		"prevAddr", caller.String(),
 		"prevRealm", previousRealm.PkgPath(),
-		"newDevOpsPct", strconv.FormatInt(newDevOpsPct, 10),
-		"prevDevOpsPct", strconv.FormatInt(prevDevOpsPct, 10),
+		"newDevOpsPct", strconv.FormatInt(state.DevOpsPct(), 10),
+		"prevDevOpsPct", strconv.FormatInt(prevState.DevOpsPct(), 10),
 		"newGovStakerPct", strconv.FormatInt(newGovStakerPct, 10),
-		"prevGovStakerPct", strconv.FormatInt(prevGovStakerPct, 10),
+		"prevGovStakerPct", strconv.FormatInt(prevState.GovStakerPct(), 10),
 	)
```
[the remark this answers is not in the document]
Comment (notJoon) [reply contract/r/gnoswap/protocol_fee/v1/protocol_fee.gno]: https://github.com/gnoswap-labs/gnoswap/pull/1093/commits/e5194b8d9d1dfe1a8b8ead580fe60e89e45e283b

File: contract/r/gnoswap/protocol_fee/v1/protocol_fee.gno [hunk 2]
```diff
@@ -103,29 +107,29 @@ func (pf *protocolFeeV1) DistributeProtocolFee() map[string]int64 {
...
+		"prevAddr", caller.String(),
 		"prevRealm", previousRealm.PkgPath(),
 		"newDevOpsPct", strconv.FormatInt(newDevOpsPct, 10),
-		"prevDevOpsPct", strconv.FormatInt(prevDevOpsPct, 10),
-		"newGovStakerPct", strconv.FormatInt(newGovStakerPct, 10),
-		"prevGovStakerPct", strconv.FormatInt(prevGovStakerPct, 10),
+		"prevDevOpsPct", strconv.FormatInt(prevState.DevOpsPct(), 10),
+		"newGovStakerPct", strconv.FormatInt(state.GovStakerPct(), 10),
+		"prevGovStakerPct", strconv.FormatInt(prevState.GovStakerPct(), 10),
 	)
```
[the remark this answers is not in the document]
Comment (notJoon) [reply contract/r/gnoswap/protocol_fee/v1/protocol_fee.gno]: https://github.com/gnoswap-labs/gnoswap/pull/1093/commits/e5194b8d9d1dfe1a8b8ead580fe60e89e45e283b

File: contract/r/gnoswap/protocol_fee/proxy.gno [hunk 3]
```diff
@@ -54,7 +56,7 @@ func GetGovStakerPct() int64 {
 }
 
 // GetTokenListWithAmount returns the list of tokens with their amounts.
-func GetTokenListWithAmount() map[string]int64 {
+func GetTokenListWithAmount() *avl.Tree {
```
[the remark this answers is not in the document]
Comment (notJoon) [reply contract/r/gnoswap/protocol_fee/proxy.gno]: https://github.com/gnoswap-labs/gnoswap/pull/1093/commits/da9675732993868ac6bc6e19f21854baf2efc50d

**Comment by jinoosss** [comment]
Currently, changing to avl.Tree is pointless as we need to iterate over all tokens.
